Добавил обработку POST запросов на WEB- роут /newsletter и отправку сообщений в Telegram всем зарегистрированным юзерам

KornilovVl opened the pull request:
Улучшил файловую структуру, повысил реюзабельность кода
Добавил обработку POST запросов на WEB- роут /newsletter и валидацию данных
Добавил функцию получения из БД всех зарегистрированных пользователей
Добавил функцию отправки сообщения в Telegram по chat Id
Подправил ридми

KornilovVl committed:
Изменил файловую структуру для повышения переиспользования кода
Вынес контроллер из cmd/api/main.go в отдельный класс
Добавил логику отправки сообщений всем зарегистрированным пользователям в telegram

KornilovVl commented:
![](https://github.trello.services/images/mini-trello-icon.png) [KU-5 Логика отправки сообщений в telegram](https://trello.com/c/gM39IsfK/6-ku-5-%D0%BB%D0%BE%D0%B3%D0%B8%D0%BA%D0%B0-%D0%BE%D1%82%D0%BF%D1%80%D0%B0%D0%B2%D0%BA%D0%B8-%D1%81%D0%BE%D0%BE%D0%B1%D1%89%D0%B5%D0%BD%D0%B8%D0%B9-%D0%B2-telegram)

wildb-ee reviewed (approved):
прочитал весь код, вроде никаких замечаний нет.

KornilovVl committed:
Изменил парсинг API- request с Form на Json
Изменил настройки докера:
Переименовал сервисы и контейнеры для более полного описания
Добавил параметры сети для связи с фронтендом
Убрал проброс порта 8080 на хост, т.к. запросы идут только из внутренней докер- сети

KornilovVl commented:
HOTFIX!
Обнаружил что  PHP сервис шлет запрос в формате json, а не в формате Form, соответственно messenger его не парсил.
Переделал парсинг запроса. По пути добавил парсинг переменной Token (чет разогнался, но уже не выпиливать же :D )
Обновил конфигурацию docker-compose: добавил настройки сети для связи с WEB сервисом, выпилил проброс портов API (обращение только через внутреннюю docker-network), изменил пробрасываемый порт БД для избежания конфликта.
Протестировал связку админка-апи, теперь все гуд

tungatarovKl commented on internal/controllers/NewsletterController.go:
```diff
@@ -0,0 +1,53 @@
+
+		//Send message for all registered users
+		for _, user := range users {
+			bot.SendMessage(user.TelegramId, apiRequest.Message)
```
вот тут можем отправлять в горутинах чтобы ускорить рассылку сообщений. После цикла, поставив waitGroup можно подождать что все отправились. Изначально можно создать waitGroup с кол-во users

KornilovVl committed:
Добавил многопоточность при отправке сообщений и тест подключения к БД

KornilovVl committed:
HOTFIX
Исправил баг многопоточности
Перенес тест БД в директорию tests для более удобного запуска тестов

KornilovVl committed:
Добавил обработку ошибок отправки
Добавил в конфиги бота timeout запроса (дефолтный был 1 минута)
